Magento2/Sniffs/Functions/DeprecatedFunctionSniff.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento2\Sniffs\Functions;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Sniff to validate PHP deprecated function.
+ */
+class DeprecatedFunctionSniff implements Sniff
+{
+    /**
+     * Deprecated functions without argument.
+     *
+     * @var array
+     */
+    private $deprecatedFunctions = [
+        'mb_check_encoding'
+    ];
+
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_OPEN_TAG
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function process(File $phpcsFile, $stackPtr): void
```

Comment:
A good approach for phpcs tests (that also matches the design of this static testing library) is utilizing the iteration that is already done by the library instead of registering T_OPEN_TAG to process every file and performing an additional custom parsing/iteration of the file.

Please see an example of the implementation where each invoked relevant token is analyzed:

```
/**
     * Deprecated functions without argument.
     *
     * @var array
     */
    private $deprecatedFunctions = [
        'mb_check_encoding' => 'Calling function mb_check_encoding() without argument is deprecated in PHP 8.1. '
            . 'Please pass the input to validate as the first argument of the function.'
    ];

    /**
     * @inheritdoc
     */
    public function register(): array
    {
        return [
            T_OPEN_PARENTHESIS
        ];
    }

    /**
     * @inheritdoc
     */
    public function process(File $phpcsFile, $stackPtr): void
    {
        $closeParenthesisPtr = $phpcsFile->findNext(T_CLOSE_PARENTHESIS, $stackPtr);
        $arguments = trim($phpcsFile->getTokensAsString($stackPtr + 1, $closeParenthesisPtr - $stackPtr - 1));
        if ($arguments) {
            return;
        }

        $functionName = $phpcsFile->getTokensAsString($phpcsFile->findPrevious(T_STRING, $stackPtr), 1);

        if (isset($this->deprecatedFunctions[$functionName])) {
            $phpcsFile->addWarning($this->deprecatedFunctions[$functionName], $stackPtr, 'DeprecatedFunction');
        }
    }
```